draft/ZEP0008.md

```diff
@@ -0,0 +1,684 @@
+    e.g. `s3+https://s3.amazonaws.com/bucket/path`.
+
+  The appropriate credentials (and bucket region for non-anonymous
+  access) must be determined automatically.
+
+  For most operations, e.g. to GET a single object, this ambiguity
+  does not matter, but for LIST operations the implementation must
+  determine the path to the root of the bucket, e.g. by initially
+  attempting the LIST operation with both possible paths and checking
+  which one succeeds.  For subsequent operations on the same endpoint
+  the result can be cached to avoid overhead.
+
+  If the path is empty or ends with `/`, the resource kind is known
+  syntatically to be a `directory`.
+
+  Otherwise, the resultant resource kind may be either `file` or
+  `directory`.
+
+  For the purpose of relative URLs, the path component includes the `bucket/`
+  prefix even if the endpoint is in fact a multi-bucket endpoint.
+
+- `gs://bucket/path/within/bucket` for Google Cloud Storage (GCS)
+
+  If the path is empty or ends with `/`, the resultant resource kind
+  is known syntatically to be a `directory`.
+
+  Otherwise, the resultant resource kind may be either `file` or
+  `directory`.
+
+The following `adapter` URL schemes are defined:
+
+- `zip:path/within/zip` for ZIP archive format
+
+  The base URL must refer to a `file` resource (which is expected to
+  be in ZIP format).
+
+  If the path is empty or ends with `/`, the resultant resource kind
+  is known syntatically to be a `directory`.
+
+  Otherwise, the resultant resource kind may be either `file` or
+  `directory`.
+
+- `ocdbt:` for [OCDBT](https://google.github.io/tensorstore/kvstore/ocdbt/index.html)
+
+  The base URL must refer to a `directory` resource (which is expected
+  to be in OCDBT format).
+
+  The URL syntax is `ocdbt:path` or `ocdbt:@version/path`, where
+  `version` is either `v123` or `2025-01-01T01:23:45.678Z`.
+
+  For example:
+
+  - `file:///tmp/dataset.ocdbt/|ocdbt:@v1/path/within/database`
+  - `file:///tmp/dataset.ocdbt/|ocdbt:path/within/database`
+  - `file:///tmp/dataset.ocdbt/|ocdbt:@2025-01-01T01:23:45.678Z/path/within/database`
+
+  While `@` is normally allowed within the path component of a URL,
+  with the `ocdbt:` URL scheme, if the path starts with `@` it must be
+  percent-encoded as `%40` to avoid ambiguity with the `@version`
+  component.  For example, a path of `@abc` can be specified as
+  `ocdbt:%40abc`.
+
+  The resultant resource kind may be either `file` or `directory`.
+
+  For the purpose of relative URLs, the path component does not include the
+  `@version/` prefix if present.
+
+- `icechunk:` for [Icechunk](https://icechunk.io/en/latest/)
+
+  The base URL msut refer to a `directory` resource (which is expected
+  to contain an Icechunk database).
+
+  The following syntaxes are supported:
+
+  - `icechunk:path/to/node/`
+  - `icechunk:@branch.BRANCH/path/to/node/`
+  - `icechunk:@tag.TAG/path/to/node/`
+  - `icechunk:@SNAPSHOT/path/to/node/`
```

Comment:
How do I point to an Icechunk repo, without a specific node? Is it something like `file:///foo/bar/icechunk:`

Reply:
> How do I point to an Icechunk repo, without a specific node? Is it something like `file:///foo/bar/icechunk:`


`file:///foo/bar/|icechunk:`

would refer to the root of the main branch.

Reply:
Can we add this as an example, it's probably the most common thing people will want to do.